cumulus/parachains/runtimes/testing/penpal/src/xcm_config.rs

```diff
@@ -42,26 +43,32 @@ use pallet_assets::Instance1;
 use pallet_xcm::XcmPassthrough;
 use polkadot_parachain_primitives::primitives::Sibling;
 use polkadot_runtime_common::impls::ToAuthor;
+use sp_io::hashing::blake2_256;
 use sp_runtime::traits::Zero;
+use sp_std::{collections::btree_set::BTreeSet, ops::ControlFlow};
 use testnet_parachains_constants::rococo::snowbridge::EthereumNetwork;
 use xcm::latest::prelude::*;
 use xcm_builder::{
 	AccountId32Aliases, AllowExplicitUnpaidExecutionFrom, AllowKnownQueryResponses,
 	AllowSubscriptionsFrom, AllowTopLevelPaidExecutionFrom, AsPrefixedGeneralIndex,
-	ConvertedConcreteId, EnsureXcmOrigin, FixedWeightBounds, FrameTransactionalProcessor,
-	FungibleAdapter, FungiblesAdapter, IsConcrete, LocalMint, NativeAsset, NoChecking,
-	ParentAsSuperuser, ParentIsPreset, RelayChainAsNative, SiblingParachainAsNative,
-	SiblingParachainConvertsVia, SignedAccountId32AsNative, SignedToAccountId32,
-	SovereignSignedViaLocation, StartsWith, TakeWeightCredit, TrailingSetTopicAsId,
-	UsingComponents, WithComputedOrigin, WithUniqueTopic,
+	ConvertedConcreteId, CreateMatcher, EnsureXcmOrigin, FixedWeightBounds,
+	FrameTransactionalProcessor, FungibleAdapter, FungiblesAdapter, IsConcrete, LocalMint,
+	MatchXcm, NativeAsset, NoChecking, ParentAsSuperuser, ParentIsPreset, RelayChainAsNative,
+	SiblingParachainAsNative, SiblingParachainConvertsVia, SignedAccountId32AsNative,
+	SignedToAccountId32, SovereignSignedViaLocation, StartsWith, TakeWeightCredit,
+	TrailingSetTopicAsId, UsingComponents, WithComputedOrigin, WithUniqueTopic,
+};
+use xcm_executor::{
+	traits::{ConvertLocation, JustTry, Properties, ShouldExecute, WeightTrader},
+	XcmExecutor,
 };
-use xcm_executor::{traits::JustTry, XcmExecutor};
 
 parameter_types! {
 	pub const RelayLocation: Location = Location::parent();
-	pub const RelayNetwork: Option<NetworkId> = None;
+	pub const RelayNetwork: NetworkId = Rococo;
```

Comment:
We cannot make this change because the https://github.com/polkadot-fellows/runtimes/pull/130 uses this parachain for testing as well. Perhaps we can set the Relay chain more generically?

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/114/commits/68abf04e4c07ea0711334df794a575917e25dd7a

With more bridge features like Transact added to penpal, we do need several configurations network binding, I've changed it to `Storage` as committed above but I think maybe it's better to separate runtime for different networks(e.g. penpal-rococo|penpal-polkadot)?

Reply:
Perhaps, yes. But since this code belongs to Parity maybe it would be better to ask someone like @bkontur.

Reply:
previously we used hack like:
```
pub storage RelayNetwork: Option<NetworkId> = None;
```
but there is also new parameters pallet, so I would suggest to go configurable way instead of hard-code Rococo to penpal, because we use it for fellows also

Reply:
well, Adrian already changed that to `storage` stuff with default to Rococo: https://github.com/paritytech/polkadot-sdk/pull/3695/files#diff-37d844ab36f3c061d846240105ddb24c0832ca0f7da21a89ea3e051c6d978b3bR64 (will be merged soon)

Reply:
@bkontur Thanks for the reminder. Does that mean I can stick to penpal with the customization required for the Ethereum bridge features?

e.g. For https://github.com/Snowfork/polkadot-sdk/pull/116 transact from the reverse direction I still use penpal for the integration/demonstration.